src/commands/infra/generate.ts

```diff
@@ -0,0 +1,190 @@
+import child_process from "child_process";
+import { getMaxListeners } from "cluster";
+import commander from "commander";
+import fs, { chmod } from "fs";
+import * as os from "os";
+import path from "path";
+import simpleGit from "simple-git/promise";
+import { logger } from "../../logger";
+
+const spkTemplatesPath = os.homedir() + "/.spk/templates";
+const git = simpleGit();
+
+/**
+ * Adds the init command to the commander command object
+ *
+ * @param command Commander command object to decorate
+ */
+export const generateCommandDecorator = (command: commander.Command): void => {
+  command
+    .command("generate")
+    .alias("g")
+    .description("Generate scaffold for terraform cluster deployment.")
+    .option(
+      "-p, --project <path to project folder to generate> ",
+      "Location of the definition.json file that will be generated"
+    )
+    .action(async opts => {
+      try {
+        if (opts.project) {
+          logger.info(
+            `spk will generate the following project: ${opts.project}`
+          );
+        } else {
+          opts.project = process.cwd();
+          logger.warn(
+            `No project folder was provided, spk will generate the current folder as a project`
+          );
+        }
+        await validateDefinition(opts.project);
+        const jsonSource = await validateTemplateSource(opts.project);
+        await validateRemoteSource(jsonSource);
+      } catch (err) {
+        logger.error(
+          "Error occurred while generating project deployment files"
+        );
+        logger.error(err);
+      }
+    });
+};
+
+/**
+ * Checks if definition.json is present locally to provided project path
+ *
+ * @param projectPath Path to the definition.json file
+ */
+export const validateDefinition = async (
+  projectPath: string
+): Promise<boolean> => {
+  try {
+    // If templates folder does not exist, create cache templates directory
```

Comment:
Why does this need to be created here?  Seems to have no impact on the functionality itself.

Reply:
I had this originally in the decorator but it needs to be inside an async function I believe. We alleviate the user having to even care about maintaining the cache template folder so generate creates it for them. I can also see the entire `/.spk` cache folder as something teams can use t share cli environments maybe. Thoughts?

Reply:
K